### PR TITLE
Fix #6521: Main module function calls shouldn't use project namespace

### LINKED ISSUE #6521: Renaming the project is breaking the use of collapsed function
### Discord username

_No response_

### What type of issue is this?

Transient – Occurring only once

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

https://user-images.githubusercontent.com/12892578/235883040-8a21812e-d56f-4947-979c-7e7a8fc61467.mov

### Expected behaviour

Renaming the project should not break the user workflow.

### How we can reproduce it?

_No response_

### Screenshots or screencasts

_No response_

### Enso Version

Nightly 03.05

### Browser or standalone distribution

Standalone distribution

### Browser Version or standalone distribution

standalone

### Operating System

MacOS

### Operating System Version

_No response_

### Hardware you are using

_No response_

### PR DESCRIPTION
**Pull request by Procrat**
### Pull Request Description

Fixes #6521: Picking a function from the CB that's defined in the main module now resolves to `Main.<func-name>` instead of `<project-name>.<func-name>`.

Note that, when collapsing nodes to a function, this referral style was already used, so this is just a change in the behaviour of the CB.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui/suggestion-database/src/entry.rs [hunk 1]
```diff
@@ -432,7 +432,7 @@ impl Entry {
 
     fn code_with_static_this(&self) -> String {
         if let Some(self_type) = &self.self_type {
-            format!("{}{}{}", self_type.alias_name(), opr::predefined::ACCESS, self.name)
+            format!("{}{}{}", self_type.name(), opr::predefined::ACCESS, self.name)
```
Comment (farmaazon) [inline app/gui/suggestion-database/src/entry.rs]: Unfortunately, it won't work with methods from Main modules of other projects. We need to check if the entry is actually from a currently opened project.

**Review by farmaazon — CHANGES_REQUESTED** [review]

File: app/gui/suggestion-database/src/entry.rs (hunk 1 above)
Comment (Procrat) [reply app/gui/suggestion-database/src/entry.rs]: Good catch. I've changed it. Do you want to have another look?

File: app/gui/src/controller/searcher.rs [hunk 2]
```diff
@@ -394,6 +394,8 @@ pub struct Filter {
     pub pattern: ImString,
     /// Additional context. A string representation of the edited accessor chain.
     pub context: Option<ImString>,
+    /// The name of the currently active module.
+    module_name: Rc<QualifiedName>,
```
Comment (farmaazon) [inline app/gui/src/controller/searcher.rs]: Please add a short info why `module_name` affects the filtering, perhaps with reference to `update_matching_info` or `code_with_static_this`.

**Review by farmaazon — APPROVED** [review]
I don't see any changes in `required_imports` method: please check if we don't add unnecessary `import local.Unnamed` import when inserting function from Main.

**Comment by Procrat** [comment]
> I don't see any changes in `required_imports` method: please check if we don't add unnecessary `import local.Unnamed` import when inserting function from Main.

@farmaazon: You were right, it was adding that import. Care to have another look before we move on to QA?

**Review by farmaazon — APPROVED** [review]
Looks good, let's start QA

**Comment by MichaelMauderer** [comment]
:green_circle: QA passed

**Pull request merged**